org.eclipse.search/search/org/eclipse/search/internal/ui/SearchDialog.java

```diff
@@ -434,13 +434,11 @@ protected Control createPageArea(Composite parent) {
 			if (imageDesc != null)
 				item.setImage(imageDesc.createImage());
 
-			if (i == fCurrentIndex) {
-				Control pageControl= createPageControl(folder, descriptor);
-				pageControl.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
-				item.setControl(pageControl);
-				fCurrentPage= descriptor.getPage();
-				fDialogSettings.put(STORE_PREVIOUS_PAGE, descriptor.getId());
-			}
+			Control pageControl = createPageControl(folder, descriptor);
```

Comment:
I am concerned that calling createPageControl will activate every plug-in that contributes a search page rather than deferring such activation until the tab is actually requested.

Reply:
Hi Jonah, thank you for taking a look at this.
I tried adding Control.requestLayout() on the tab folder, the control from createPageControl(...), the parent in createPageArea() and the active shell, but none of these had an effect.
I noticed the problem presented itself until all the tab content had been created, after which the white artifacts no longer appeared. Hence the rational for creating them all at the start.
However you make a good point about deferred activation. If this is unacceptable then I'll go back to the drawing board with this.

Reply:
Unless something has changed about the general requirements about activation I would guess that having a significant change of activation will be a problem.

Looking at this code a little bit it looks like the issue only happens in relation to the code if `showScope` is true - and in that block if you don't call `fScopeParts[index].setVisible(true);` then the problem goes away. The setVisible calls layout on the control, so it looks to me like a bug the first time layout is called. If so, that indicates there is some underlying bug that probably applies to numerous layouts. I guess the best case is to extract what that set of layouts are and see if it can be reproduced in a smaller set as that would fix this type of issue in multiple places in one go.

Note this is a place the SWT and layout spy can be problematic as it can display the result of current calculations, not the results that were actually used to display the gui.

Reply:
Hi,

As per @jonahgraham's suggestion, I've created a compact reproduction test case which demonstrates this problem.


The file Bug552728.zip is an Eclipse Plug-in Project. It contains a compact test reproduction case to demonstrate the bug:
Bug 552728 - [Dark Theme] Search shows white artifacts
To run test:
* Place the Bug552728 plugin in an Eclipse SDK workspace and launch a runtime workbench.
* In Preference > General > Appearance, change Theme to Dark and restart.
* In the main toolbar, select the menu Bug552728 and choose SearchDialogTest (or press Ctrl+6).
* A modal dialog appears containing 3 tab items, with item 0 selected.
* Click Item 1 tab:
Expected: The text "Content for Item" is shown in white, the remaining background area is drawn in the dark theme background colour.
Actual: The text "Content for Item" is shown in white with a small dark background and the remaining background area is drawn in white.


I've stepped through Composite.layout() and GridLayout.layout(Composite, boolean) and don't see anything obviously suspicious, although the code is insanely complex.

I would really appreciate it if someone familiar with SWT could take a look please.

Cheers John

[Bug552728.zip](https://github.com/eclipse-platform/eclipse.platform.text/files/10842484/Bug552728.zip)